### PR TITLE
Assorted render fixes

### PR DESCRIPTION
**Pull request by hrydgard**
Fix a bunch of minor bugs (well, the GL texture filter one is pretty bad) and do some cleanups.

Also unifies a function, BlitFramebufferDepth.

Just stuff I've run into while investigating #13431.

### CODE REVIEW

File: GPU/D3D11/GPU_D3D11.cpp [hunk 1]
```diff
@@ -146,11 +146,11 @@ void GPU_D3D11::CheckGPUFeatures() {
 		features |= GPU_SUPPORTS_DUALSOURCE_BLEND;
 	if (draw_->GetDeviceCaps().depthClampSupported)
 		features |= GPU_SUPPORTS_DEPTH_CLAMP;
-	features |= GPU_SUPPORTS_ANY_COPY_IMAGE;
+	features |= GPU_SUPPORTS_COPY_IMAGE;
 	features |= GPU_SUPPORTS_TEXTURE_FLOAT;
 	features |= GPU_SUPPORTS_INSTANCE_RENDERING;
 	features |= GPU_SUPPORTS_TEXTURE_LOD_CONTROL;
-	features |= GPU_SUPPORTS_FBO;
+	features |= GPU_SUPPORTS_FRAMEBUFFER_BLIT;
```
Comment (aliaspider) [inline GPU/D3D11/GPU_D3D11.cpp:153]: isn't `BlitFramebuffer` stubbed for `d3d11` ?
https://github.com/hrydgard/ppsspp/blob/master/ext/native/thin3d/thin3d_d3d11.cpp#L1414-L1419

File: GPU/Common/FramebufferManagerCommon.cpp [hunk 2]
```diff
@@ -461,6 +469,26 @@ void FramebufferManagerCommon::DestroyFramebuf(VirtualFramebuffer *v) {
...
+		int w = std::min(src->renderWidth, dst->renderWidth);
+		int h = std::min(src->renderHeight, dst->renderHeight);
+		// Let's only do this if not clearing depth.
+		if (gstate_c.Supports(GPU_SUPPORTS_FRAMEBUFFER_BLIT)) {
+			draw_->BlitFramebuffer(src->fbo, 0, 0, w, h, dst->fbo, 0, 0, w, h, Draw::FB_DEPTH_BIT, Draw::FB_BLIT_NEAREST, "BlitFramebufferDepth");
+			RebindFramebuffer("BlitFramebufferDepth");
+		} else if (gstate_c.Supports(GPU_SUPPORTS_COPY_IMAGE)) {
+			draw_->CopyFramebufferImage(src->fbo, 0, 0, 0, 0, dst->fbo, 0, 0, 0, 0, w, h, 1, Draw::FB_DEPTH_BIT, "BlitFramebufferDepth");
+			RebindFramebuffer("BlitFramebufferDepth");
+		}
```
Comment (aliaspider) [inline GPU/Common/FramebufferManagerCommon.cpp:487]: does `BlitFramebuffer` here have higher priority than `CopyFramebufferImage` on purpose ?
`CopyFramebufferImage` should be faster no ?

Comment (aliaspider) [reply GPU/Common/FramebufferManagerCommon.cpp:487]: nvm just saw the comment above :).

File: GPU/D3D11/GPU_D3D11.cpp (hunk 1 above)
Comment (hrydgard) [reply GPU/D3D11/GPU_D3D11.cpp:153]: Right you are, oops.

File: GPU/Common/FramebufferManagerCommon.cpp (hunk 2 above)
Comment (hrydgard) [reply GPU/Common/FramebufferManagerCommon.cpp:487]: Actually not sure if that comment applies to all the backends, I think it got copy pasted from the GL backend..

**Pull request merged**